packages/@uppy/store-default/src/index.ts

```diff
@@ -0,0 +1,53 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore We don't want TS to generate types for the package.json
+import packageJson from '../package.json'
+
+export type StateOrStateFragment = Record<string, unknown>
+
+export type Listener = (
+  prevState: StateOrStateFragment,
+  nextState: StateOrStateFragment,
+  patch: StateOrStateFragment,
+) => void
+
+/**
+ * Default store that keeps state in a simple object.
+ */
+class DefaultStore {
+  static VERSION = packageJson.version
+
+  public state: StateOrStateFragment
```

Comment:
I think `DefaultStore` should accept a generic which makes the state strongly typed. Otherwise we have to type check _everything_ and have no editor intelligence.